java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/db/DBEntityFromTables.java

```diff
@@ -0,0 +1,139 @@
+import org.netbeans.api.progress.aggregate.ProgressContributor;
+import org.netbeans.api.project.Project;
+import org.netbeans.api.project.ProjectUtils;
+import org.netbeans.api.project.SourceGroup;
+
+import org.netbeans.modules.j2ee.persistence.api.entity.generator.EntitiesFromDBGenerator;
+import org.netbeans.modules.java.lsp.server.LspServerState;
+import org.netbeans.modules.java.lsp.server.protocol.CodeActionsProvider;
+import org.netbeans.modules.java.lsp.server.protocol.NbCodeLanguageClient;
+import org.netbeans.modules.java.lsp.server.protocol.QuickPickItem;
+import org.netbeans.modules.java.lsp.server.protocol.ShowInputBoxParams;
+import org.netbeans.modules.java.lsp.server.protocol.ShowQuickPickParams;
+import org.netbeans.modules.parsing.api.ResultIterator;
+import org.openide.util.Lookup;
+import org.openide.util.NbBundle;
+import org.openide.util.lookup.ServiceProvider;
+
+/**
+ *
+ * @author Jan Horvath
+ */
+@NbBundle.Messages({
+    "MSG_NoProject=No Project Open",
+    "MSG_NoDbConn=No DB Connection",
+    "MSG_NoSourceRoot=No source root found",
+    "MSG_SelectTables=Select Database Tables",
+    "MSG_EnterPackageName=Enter package name"
+})
+@ServiceProvider(service = CodeActionsProvider.class)
+public class DBEntityFromTables extends CodeActionsProvider {
+
+    private static final String COMMAND_ENTITY_FROM_TABLES = "db.entity.from.tables"; //NOI18N
+
+    @Override
+    public CompletableFuture<Object> processCommand(NbCodeLanguageClient client, String command, List<Object> arguments) {
+        if (!COMMAND_ENTITY_FROM_TABLES.equals(command)) {
+            return null;
+        }
+        Lookup.getDefault().lookup(LspServerState.class).openedProjects().thenAccept((projects) -> {
+            if (projects.length > 0) {
+                createEntityClassesInProject(client, projects[0]);
+            } else {
+                client.showMessage(new MessageParams(MessageType.Error, Bundle.MSG_NoProject()));
+            }
+
+        });
+        return null;
+    }
+
+    private CompletableFuture createEntityClassesInProject(NbCodeLanguageClient client, Project prj) {
+        try {
+            DatabaseConnection connection = ConnectionManager.getDefault().getPreferredConnection(true);
+            if (connection == null) {
+                client.showMessage(new MessageParams(MessageType.Error, Bundle.MSG_NoDbConn()));
+                return null;
+            }
+            ConnectionManager.getDefault().connect(connection);
+            SourceGroup[] sr = ProjectUtils.getSources(prj).getSourceGroups(JavaProjectConstants.SOURCES_TYPE_JAVA);
+            if (sr.length < 1) {
+                client.showMessage(new MessageParams(MessageType.Error, Bundle.MSG_NoSourceRoot()));
+                return null;
+            }
+            Connection conn = connection.getJDBCConnection();
+            ResultSet rs = conn.getMetaData().getTables(conn.getCatalog(), conn.getSchema(), "%", new String[]{"TABLE", "VIEW"}); //NOI18N
```

Comment:
This call eventually waits for the data from the JDBC connection; I'd suggest to run the DB query in a RequesProcessor so the LSP communication is not blocked. The behaviour depends on the DB location / network conditions.